src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/chapterPage/viewModel/ChapterPageViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package org.wycliffeassociates.otter.jvm.app.ui.chapterPage.viewModel
+
+import io.reactivex.rxkotlin.subscribeBy
+import javafx.collections.FXCollections
+import org.wycliffeassociates.otter.jvm.app.ui.chapterPage.model.ChapterPageModel
+import org.wycliffeassociates.otter.jvm.app.ui.chapterPage.model.Verse
+import org.wycliffeassociates.otter.jvm.app.ui.inject.Injector
+import tornadofx.*
+
+class ChapterPageViewModel : ViewModel() {
+    val model = ChapterPageModel()
+//    val chapterPageUseCase = ChapterPageUseCase(Injector.chapterDao)
+    private val modelChapters = model.chapters
+    val activeChapter = model.activeChapter
+    val selectedTab = model.selectedTab
+    val chapters = FXCollections.observableArrayList<String>()!!
+    val verses = FXCollections.observableArrayList<Verse>()!!
+    val bookTitle = model.bookTitle
+
+    fun changeContext(context: String) {
+        selectedTab.onNext(context)
+    }
+
+    init {
+        mapChapters()
+    }
+
+    private fun mapChapters() {
```

Comment:
Just a personal preference, but it might be more readable to name this function something like `generateChapterHeaders` to stress what it is doing

Reply:
however it's doing more that just generating chapter headers. It's also houses the observer which controls which verses are visible. so maybe something like `initializeChapterPage?`

Reply:
Yep, that would work!